doc/index.md

```diff
@@ -1,61 +1,119 @@
+local cq = cqueues.new()
 
-A ["stream"](#stream) is a request/response on a connection.
-lua-http has two stream types: one for [HTTP 1 streams](#http.h1_stream), and one for [HTTP 2 streams](#http.h2_stream).
-They share a lowest common denominator interface, see [*stream*](#stream) and [*http.stream_common*](#http.stream_common).
+local shutdown = false
+local ws_ok, err, errno = ws:connect()
+if ws_ok then
 
 
-## Common use cases
+    cq:wrap(function()
+        local ok, err, errno = ws:send("Hidie Ho Neighbour!")
 
-### Retrieving a document
+        if not ok then
+            io.stderr:write(string.format("%s: %s %s\n", err, errno, debug.traceback()))
+            os.exit(1)
+        end
+    end)
 
-The highest level interface for clients is [*http.request*](#http.request). By constructing a [*request*](#http.request) object from a uri using [`new_from_uri`](#http.request.new_from_uri) and immediately evaluating it, you can easily fetch an HTTP resource.
+    cq:wrap(function()
+        repeat
+            print("receiving...")
 
-```lua
-local http_request = require "http.request"
-local headers, stream = assert(http_request.new_from_uri("http://example.com"):go())
-local body = assert(stream:get_body_as_string())
-if headers:get ":status" ~= "200" then
-	error(body)
+            local response, err, errno = ws:receive()
+            if not response then
+                io.stderr:write(string.format("%s: %s %s %s\n", err, errno, "Receive failed", debug.traceback()))
+            else
+                print("Received: " .. response .. "- sizeof: " .. #response)
+            end
+        until shutdown == true
+    end)
+
+    assert(cq:loop())
+else
+    io.stderr:write(string.format("%s: %s %s %s\n", err, errno, "Failed to connect.", debug.traceback()))
 end
-print(body)
 ```
 
-
 ## Asynchronous Operation
 
-All lua-http operations including DNS lookup, socket connection, TLS negotiation and read/write operations are all asynchronous when run inside of a cqueue.
-[Cqueues](http://25thandclement.com/~william/projects/cqueues.html) is a lua library that allows for composable event loops.
-Cqueues can be integrated with almost any main loop or event library you may encounter (see [here](https://github.com/wahern/cqueues/wiki/Integrations-with-other-main-loops) for more information + samples), and hence lua-http can be asynchronous in any place you write lua!
+lua-http has been written to perform asyncrhounously to greatly reduce resource overhead in an application. Asynchronous operations are achieved by utilizing cqueues, a Lua/C library that incorporates Lua yielding and kernel level APIs to reduce CPU usage. Most lua-http operations including DNS lookup, TLS negotiation and read/write operations are asynchronous. While sometimes it is necessary to block a thread, any operations that may block the current coroutine take an optional timeout to ensure your application does not hang in an unusable state.
```

Comment:
"While sometimes it is necessary to block a thread" => a lua **thread** is always blocked by an operation. however your whole application/main loop should never be.

(the timeout is so that an operation doesn't stay pending forever if a response never comes: good networking code should always assume that the other end will not reply/crash/etc.)